### ISSUE TITLE
`jsx-quotes`: should be in jsx package, not js

### ISSUE THREAD

**Issue by wJoenn** [eslint-stylistic/eslint-stylistic#303]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

The `jsx-rules` is part of the js package.
https://eslint.style/rules/js/jsx-quotes

I don't really see a reason why it shouldn't be in the jsx package instead

### Reproduction

Shouldn't be needed, the issue is obvious enough

### Contributes

- [X] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by antfu** [comment]
It's from ESLint core, we keep that for backward compatibility and easier migration.

**Comment by wJoenn** [comment]
I noticed you renamed one of the rules from `func-call-spacing` to `function-call-spacing`
https://eslint.style/rules/js/func-call-spacing

If it's a pattern that you've used already, have you considered doing something similar with this rule too ? As in moving the rule but keep the existing readme that redirects to the new one.
The old one could still work but with a deprecation warning too or something along this idea.

Or are you reluctant because it's not simply renaming a rule within the same package but moving the rule from one package to another ?

**Comment by antfu** [comment]
Yeah, I wouldn't mind moving it and alias it back. Pr welcome if that matters to you.
